File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size()==1) return 1;
        if (nums.size()==2 && nums[0]==nums[1]) return 1; 

        int i = 0;
        while (i<nums.size()-1) {
            if (nums[i]==nums[i+1]) {
                int idx = i+1;
                while (nums[idx]==nums[i] && idx<nums.size()) {
                    idx++;
                }
                nums.erase(nums.begin()+i+1, nums.begin()+idx);
            } else i++;            
        }

        return nums.size();
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return 1
        
        idx = 0
        while idx < len(nums)-1:
            if nums[idx] == nums[idx+1]:
                nums.pop(idx)
            else:
                idx += 1
        return len(nums)